content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md

```diff
@@ -175,16 +175,34 @@ For more information on version skews, see:
 
 {{< tabs name="k8s_install" >}}
 {{% tab name="Debian-based distributions" %}}
-```bash
-sudo apt-get update && sudo apt-get install -y apt-transport-https curl
-curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
-cat <<EOF | sudo tee /etc/apt/sources.list.d/kubernetes.list
-deb https://apt.kubernetes.io/ kubernetes-xenial main
-EOF
-sudo apt-get update
-sudo apt-get install -y kubelet kubeadm kubectl
-sudo apt-mark hold kubelet kubeadm kubectl
-```
+
+1. Update the `apt` package index and install packages needed to use the Kubernetes `apt` repository:
+
+   ```shell
+   sudo apt-get update
+   sudo apt-get install -y apt-transport-https ca-certificates wget
+   ```
+
+2. Download the Google Cloud public signing key:
+
+   ```shell
+   sudo wget -O /usr/share/keyrings/kubernetes-archive-keyring.gpg https://packages.cloud.google.com/apt/doc/apt-key.gpg
+   ```
+
+3. Add the Kubernetes `apt` repository:
+
+   ```shell
+   echo "deb [signed-by=/usr/share/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee /etc/apt/sources.list.d/kubernetes.list
+   ```
+
```

Comment:
How about package pinning? I'm thinking of https://wiki.debian.org/DebianRepository/UseThirdParty#Complete_example

Perhaps that's something for a follow-up issue?

Reply:
Hmm, I'm inclined to leave that to a possible follow-up. The links says a preferences file `SHOULD` be created, but is now required. Im not entirely sure if `Pin-Priority: ...` might actually conflict with `apt-mark hold` that we make use of in the next step for installing kubeadm, kubelet, and kubectl together (but not kubectl on its own - see install-kubectl-linux.md), or what value we want (1, 100, etc).

The origin hostname might also be problematic: https://apt.kubernetes.io/ is a redirect to https://packages.cloud.google.com/apt/. Im note sure which if the "original" or "followed" hostname is used; and the links makes note

> Note that if the local system pulls multiple repositories from the same host (e.g. different paths, different suites, or different components), then the proposed Pin: origin is incapable of distinguishing between them. Fixing this appears to require improvements in apt, see 858406.

Which is a problem as https://packages.cloud.google.com/apt/ is used for other repos too, e.g. [Google Cloud SDK](https://cloud.google.com/sdk/docs/install).